### PR TITLE
Additonal serverspec checks for MongoDB

### PR DESCRIPTION
**Pull request by lihkin213**

### CODE REVIEW

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb [hunk 1]
```diff
@@ -1,44 +1,72 @@
...
 
 describe file ('/etc/mongod.conf') do
-  it { should contain 'logpath=/var/log/mongodb/mongod.log' }
+  it { should contain 'logpath=/var/log/mongodb/mongod.log' } #MON005
 end
 
 describe file ('/etc/mongod.conf') do
-  it { should contain 'logappend=true' }
+  it { should contain 'logappend=true' }  #MON004
 end
```
Comment (masteinhauser) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Why not combine this and the `describe` right above it?

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb [hunk 2]
```diff
@@ -1,44 +1,72 @@
...
 
-describe file ('/var/lib/mongodb') do
-  it { should be_mode 755 }
-  it { should be_owned_by 'mongodb' }
-  it { should be_grouped_into 'mongodb'}
+describe file ('/var/lib/mongodb') do    
+  it { should be_mode 755 }               #MON008
+  it { should be_owned_by 'mongodb' }     #MON007
+  it { should be_grouped_into 'mongodb'}  #MON009
 end
```
Comment (masteinhauser) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Again, combine this and the previous `describe file` into the same.

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb [hunk 3]
```diff
@@ -1,44 +1,72 @@
...
+  it { should be_mode 644 }                 #MON016
+  it { should be_owned_by 'mongodb' }       #MON017
+  it { should be_grouped_into 'nogroup' }   #MON018
+end
+
+describe file ('/var/lib/mongodb/mongod.lock') do
+  it { should be_mode 755 }                 #MON013
+  it { should be_owned_by 'mongodb' }       #MON014
+  it { should be_grouped_into 'nogroup' }   #MON015
+end
```
Comment (masteinhauser) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Does this make sense? Won't the trigger when mongodb is *not* running for some reason? If so, it doesn't seem like this is a good check.

File: roles/mongodb-common/tasks/main.yml [hunk 4]
```diff
@@ -6,7 +6,7 @@
     - mongodb-org-shell={{ mongodb.version }}
 
 - name: install mongodb config files
-  template: src=etc/mongodb/mongod.conf dest=/etc/mongod.conf owner=mongodb mode=0640
+  template: src=etc/mongodb/mongod.conf dest=/etc/mongod.conf owner=root group=mongodb mode=0640
```
Comment (epwn) [inline roles/mongodb-common/tasks/main.yml]: should this be a separate PR since this isn't related to serverspec checks?

Comment (j2sol) [inline roles/mongodb-common/tasks/main.yml]: Well, it's setting the thing serverspec is looking for, so I'm okay with it being in here.

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb (hunk 1 above)
Comment (j2sol) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Yeah I think it should be collapsed too.

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb [hunk 5]
```diff
@@ -1,44 +1,72 @@
...
 
 files = Dir.glob("/var/lib/mongodb/local.*")
 files.each do |file|
   describe file("#{file}") do
-    it { should be_mode 600 }
-    it { should be_owned_by 'mongodb' }
-    it { should be_grouped_into 'nogroup' }
+    it { should be_mode 600 }              #MON010
+    it { should be_owned_by 'mongodb' }    #MON011
+    it { should be_grouped_into 'nogroup' } #MON012
```
Comment (j2sol) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Looks like the length of this line offsets the whitespace you're trying to get in the previous two lines.

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb (hunk 3 above)
Comment (j2sol) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Seconded, any way to skip this check if the file doesn't exist?

Comment (lihkin213) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: If mongodb is installed, should it not be running, I should probably add a check to make sure this file exists, and if it does not it should be an alert for the operations to check what is going on.

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb (hunk 1 above)
Comment (lihkin213) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: Don't know what I was thinking :-) Thanks for the catch

File: roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb (hunk 3 above)
Comment (masteinhauser) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: This is a correct assumption applied in an incorrect line of code. 

If you wanting to check if a service is running, you should be using the `sensu_process_check` module. Remember, *serverspec is only executed _once per hour_*. Do you really only want to be alerted that mongodb is not running after it could have been down for 59 (or a bit more) minutes? This is a poor check.

Comment (lihkin213) [inline roles/mongodb-common/templates/etc/serverspec/mongodb_spec.rb]: You are absolutely correct. There is already a process check for it. This makes no sense here. I will take all of it out.

**Comment by lihkin213** [comment]
Updated based on comments. Please review.

**Comment by masteinhauser** [comment]
Post review updates LGTM. 👍

**Comment by j2sol** [comment]
👍
